pkg/cmd/pulumi/state_move.go

```diff
@@ -225,21 +229,51 @@ func breakDependencies(res *resource.State, resourcesToMove map[string]*resource
 	}
 }
 
-func rewriteURNs(res *resource.State, dest backend.Stack) {
-	// TODO: rewrite project name
-	res.URN = res.URN.RenameStack(dest.Ref().Name())
+func renameStackAndProject(urn urn.URN, stack backend.Stack) (urn.URN, error) {
+	newURN := urn.RenameStack(stack.Ref().Name())
+	if project, ok := stack.Ref().Project(); ok {
+		newURN = newURN.RenameProject(tokens.PackageName(project))
+	} else {
+		return "", errors.New("cannot get project name")
```

Comment:
This is only going to be for old pre-project diy backends, maybe worth being explicit here and telling the user they need to `state upgrade`?

Reply:
Done! :heavy_check_mark: